control/controlclient/noise.go

```diff
@@ -344,7 +360,7 @@ func (nc *NoiseClient) Close() error {
 
 // dial opens a new connection to tailcontrol, fetching the server noise key
 // if not cached.
-func (nc *NoiseClient) dial() (*noiseConn, error) {
+func (nc *NoiseClient) dial(ctx context.Context) (*noiseConn, error) {
```

Comment:
document that this ctx is 1 of N waiting dialers and thus shouldn't assume any context values.